services/repository/push.go

```diff
@@ -107,7 +107,7 @@ func pushUpdates(optsList []*repo_module.PushUpdateOptions) error {
 		if opts.IsNewRef() && opts.IsDelRef() {
 			return fmt.Errorf("old and new revisions are both %s", git.EmptySHA)
 		}
-		if opts.IsTag() { // If is tag reference
+		if opts.RefFullName.IsTag() { // If is tag reference
```

Comment:
```suggestion
		if opts.RefFullName.IsTag() {
```
That comment is useless. That's literally what this if says